CHANGELOG.md

```diff
@@ -44,6 +44,16 @@ Note that 3.1.0 will be the first API stable release and interfaces in this rele
 
 ## API changes
 
+### Search
+
+* Configuration refactoring:
+  * There does not exist a `search_cfg::mode` struct anymore but individual config elements. To adapt your code change
+    the following naming:
+    `seqan3::search_cfg::mode{seqan3::search_cfg::all}` to `seqan3::search_cfg::reporting_all_hits`
```

Comment:
just one remark. We used reporting here. But in the alignment it does not make sense to call it reporting_global_alignment or reporting_local_alignment. Rather computing_local_aignment etc. Should we use this then instead? Also finding would be an alternative for reporting which would also work for alignments.

Reply:
@rrahn Makes sense. I guess `computing` is the most intuitive for both? since it matches best with any algorithm.
I would even think `compute_all_hits` is nicer than `computing_all_hits` ... how strict are we with this "verbs are for function names?"

Reply:
I am not so sure about the verb. In general it should be fine since we use brace initialisation`{}` syntax and then what about gaps or max_error? Why don't they have a verb then? I think about this when I make a plan for the dynamic and static modes. So we can discuss this together on the next stratgy meeting and hopefully resolve this as quickly.